### PR TITLE
ci: Update dependabot configuration

### PR DESCRIPTION
**Pull request by riscait**
## 🙌 What's Done

- We have used grouped to allow pull requests to be created together to some extent.

## ✍️ What's Not Done

-

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -11,33 +11,49 @@ updates:
...
-      interval: "monthly"
+    groups:
+      riverpod-dependencies:
+        patterns:
+          - "*riverpod*"
+      freezed-dependencies:
+        patterns:
+          - "freezed*"
+          - "json_annotation"
+          - "json_serializable"
```
Comment (riscait) [inline .github/dependabot.yml:31]: Multiple dependent packages, such as Riverpod, Freezed, etc., are now grouped together.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -11,33 +11,46 @@ updates:
...
+      dependencies-patch:
+        patterns:
+          - "*"
+        update-types:
+        - "patch"
```
Comment (riscait) [inline .github/dependabot.yml:36]: Patch version updates were combined into one.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -11,33 +11,46 @@ updates:
...
 
   - package-ecosystem: "pub"
     directory: "/packages/themes"
     schedule:
       interval: "weekly"
+    groups:
+      dependencies:
+        patterns:
+          - "*"
```
Comment (riscait) [inline .github/dependabot.yml:45]: The dependencies are small packages so that they can be combined into one PR.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -11,33 +11,46 @@ updates:
...
   - package-ecosystem: "pub"
     directory: "/packages/util"
     schedule:
       interval: "monthly"
+    groups:
+      dependencies-patch:
+        patterns:
+          - "*"
+        update-types:
+        - "patch"
```
Comment (riscait) [inline .github/dependabot.yml:56]: Patch perversion will be combined into one PR.

File: .github/dependabot.yml [hunk 5]
```diff
@@ -11,33 +11,46 @@ updates:
...
-
-  - package-ecosystem: "pub"
-    directory: "/packages/amazon_paapi"
-    schedule:
-      interval: "monthly"
-
-  - package-ecosystem: "pub"
-    directory: "/packages/convenient_widgets"
-    schedule:
-      interval: "monthly"
```
Comment (riscait) [inline .github/dependabot.yml:23]: Packages that were not actively maintained were excluded.

**Review by naipaka — APPROVED** [review]
The group configurator looks great 👍
LGTM

**Review by k-nkmr — APPROVED** [review]
LGTM

**Pull request merged**